middleware/authMiddleware.js

```diff
@@ -0,0 +1,33 @@
+const jwt = require("jsonwebtoken");
+const User = require("../backend/models/userModel");
+const asyncHandler = require("express-async-handler");
+
+const protect = asyncHandler(async (req, res, next) => {
+  let token;
+
+  if (
+    req.headers.authorization &&
+    req.headers.authorization.startsWith("Bearer")
+  ) {
+    try {
+      token = req.headers.authorization.split(" ")[1];
+
+      //decodes token id
+      const decoded = jwt.verify(token, process.env.JWT_SECRET);
+
+      req.user = await User.findById(decoded.id).select("-password");
+
+      next();
+    } catch (error) {
+      res.status(401);
+      throw new Error("Not authorized, token failed");
```

Comment:
Static strings should be in the constant file so that the same can be re-used and if needed to change, could easily be done. Thanks